src/components/Modal/__stories__/modal.stories.mdx

```diff
@@ -0,0 +1,308 @@
+import Modal from "../Modal";
+import { useModalHelper } from "./helpers";
+import { ArgsTable, Story, Canvas, Meta } from "@storybook/addon-docs";
+import { createStoryMetaSettings } from "../../../storybook/functions/create-component-story";
+import "./Modal.stories.scss";
+import { ModalContent, ModalFooter, ModalHeader, TextField } from "components";
+import { Settings } from "components/Icon/Icons";
+
+export const metaSettings = createStoryMetaSettings({
+  component: Modal,
+  enumPropNamesArray: [], // List enum props here
+  iconPropNamesArray: [], // List props that are typed as icons here
+  actionPropsArray: [] // List the component's actions here
+});
+
+<Meta title="Feedback/Modal" component={Modal} argTypes={metaSettings.argTypes} decorators={metaSettings.decorators} />
+
+<!--- Component documentation -->
+
+# Modal
+
+- [Overview](#overview)
+- [Props](#props)
```

Comment:
I think we have an issue with controls settings:
![Screen Shot 2022-06-26 at 12 34 22](https://user-images.githubusercontent.com/96241324/175808898-c5bd84d6-b0db-4dd3-bab8-0382b3e6eada.png)

Please check other stories for setups it - in you case I think Accordion will be good example since it also have sub-components as Modal.